enclave/src/main/java/org/hyperledger/besu/enclave/Enclave.java

```diff
@@ -25,102 +26,126 @@
   }
 
   public SendResponse send(final SendRequest content) {
-    return executePost(buildPostRequest(JSON, content, "/send"), SendResponse.class);
+    return postRequest(
+        JSON,
+        content,
+        "/send",
+        (statusCode, body) -> handleJsonResponse(statusCode, body, SendResponse.class));
   }
 
   public ReceiveResponse receive(final ReceiveRequest content) {
-    return executePost(buildPostRequest(ORION, content, "/receive"), ReceiveResponse.class);
+    return postRequest(
+        ORION,
+        content,
+        "/receive",
+        (statusCode, body) -> handleJsonResponse(statusCode, body, ReceiveResponse.class));
   }
 
   public PrivacyGroup createPrivacyGroup(final CreatePrivacyGroupRequest content) {
-    return executePost(buildPostRequest(JSON, content, "/createPrivacyGroup"), PrivacyGroup.class);
+    return postRequest(
+        JSON,
+        content,
+        "/createPrivacyGroup",
+        (statusCode, body) -> handleJsonResponse(statusCode, body, PrivacyGroup.class));
   }
 
   public String deletePrivacyGroup(final DeletePrivacyGroupRequest content) {
-    return executePost(buildPostRequest(JSON, content, "/deletePrivacyGroup"), String.class);
+    return postRequest(
+        JSON,
+        content,
+        "/deletePrivacyGroup",
+        (statusCode, body) -> handleJsonResponse(statusCode, body, String.class));
   }
 
   public PrivacyGroup[] findPrivacyGroup(final FindPrivacyGroupRequest content) {
-    Request request = buildPostRequest(JSON, content, "/findPrivacyGroup");
-    return executePost(request, PrivacyGroup[].class);
+    return postRequest(
+        JSON,
+        content,
+        "/findPrivacyGroup",
+        (statusCode, body) -> handleJsonResponse(statusCode, body, PrivacyGroup[].class));
   }
 
-  private Request buildPostRequest(
-      final MediaType mediaType, final Object content, final String endpoint) {
-    final RequestBody body;
+  private <T> T postRequest(
+      final String mediaType,
+      final Object content,
+      final String endpoint,
+      final ResponseBodyHandler<T> responseBodyHandler) {
+    final String bodyText;
     try {
-      body = RequestBody.create(mediaType, objectMapper.writeValueAsString(content));
+      bodyText = objectMapper.writeValueAsString(content);
     } catch (final JsonProcessingException e) {
-      throw new EnclaveException("Failed to serialise request to json body.");
+      throw new EnclaveException("Unable to serialise object to json representation.");
     }
-    final String url = enclaveUri.resolve(endpoint).toString();
-    return new Request.Builder().url(url).post(body).build();
-  }
 
-  private <T> T executePost(final Request request, final Class<T> responseType) {
+    return requestTransmitter.post(mediaType, bodyText, endpoint, responseBodyHandler);
+  }
 
-    final Response response;
-    final String responseBody;
+  private <T> T getRequest(
+      final String mediaType,
+      final Object content,
```

Comment:
Shouldn't need the content at all for the get request

Reply:
not the contentType.